src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -35,5 +37,39 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::RequestCommissioning(int commissionerNumber)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    const Mdns::DiscoveredNodeData * commissioner;
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        commissioner = GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr && --commissionerNumber == 0)
+        {
+            break;
+        }
+    }
+    if (commissionerNumber != 0)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Enter commissioning mode, open commissioning window
```

Comment:
The fact that `InitServer()` opens a commissioning window is temporary behavior, and when spec compliance is improved with regards to commissioning window, it's possible no commissioning window will be open at all from this call. If you desire to forcible open a commissioning window, there are other means.

Reply:
Called OpenBasicCommissioningWindow() to ensure a commissioning window is opened